src/modules/companies/companies.controller.ts

```diff
@@ -0,0 +1,28 @@
+import { Controller, Get, Query } from '@nestjs/common';
+import { ApiNotFoundResponse, ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+
+import { CompaniesService } from './companies.service';
+import { GetCompanyByRegistrationNumberQuery } from './dto/get-company-by-registration-number-query.dto';
+import { GetCompanyResponse } from './dto/get-company-response.dto';
+
+@ApiTags('companies')
+@Controller('companies')
+export class CompaniesController {
+  constructor(private readonly companiesService: CompaniesService) {}
+
+  @Get()
+  @ApiOperation({
+    summary: 'Get company by Companies House registration number.',
+  })
+  @ApiResponse({
+    status: 200,
+    description: 'Returns the company',
+    type: GetCompanyResponse,
+  })
+  @ApiNotFoundResponse({
```

Comment:
I've added documentation for `400`, `404` (if CH API returns a `404`) and `500` responses as these are possible response from our endpoint. Note that this endpoint returns a `400` if the request fails the query validation, *not* if Companies House API returns a `400`. For non-`404` responses from CH API, I think we should return a `500` from our endpoint: 
- 400: we do request validation before we send our request to CH API, so if CH API returns a 400, it should be due to a problem in MDM (e.g., we don't construct our 'Authorization' header properly in the code).
- 401: if CH API returns a 401, the most likely problem is an incorrect API key in MDM, so it is a 500 as far as DTFS is concerned.
- 422: similar to the 400, this suggests a problem in MDM, not in the request from DTFS.
- 429: I think we should have our services' rate-limits set to a sensible level to avoid hitting the rate-limits of 3rd party services. If we do get a 429 from CH API, I think it's more useful to log this error in MDM (which we do anyway) so we can adjust our rate-limit and return a 500 to the user.
- 500: a 500 from CH API clearly isn't due to the request from DTFS, so a 500 makes sense here also.

Reply:
Understod thank you for a detailed explanation.